theano/tensor/alt_sgemm.c

```diff
@@ -0,0 +1,77 @@
+/**
+C Implementation of sgemm_ based on NumPy
+Used instead of blas when Theano config flag blas.ldflags is empty.
+**/
+inline PyObject* alt_transpose(PyArrayObject* o) {
+	return PyArray_Transpose(o, NULL);
+}
+inline PyObject* alt_matrix_matrix_product(PyObject* o1, PyObject* o2) {
+	return PyArray_MatrixProduct(o1, o2);
+}
+inline PyObject* alt_matrix_matrix_product2(PyObject* o1, PyObject* o2, PyObject* out) {
+	return PyArray_MatrixProduct2(o1, o2, (PyArrayObject*)out);
+}
+void alt_scalar_matrix_product_in_place(float scalar, float* matrix, int size_to_compute) {
+	for(int i = 0; i < size_to_compute; ++i) {
+		matrix[i] *= scalar;
+	}
+}
+void alt_matrix_sum_in_place(float* A, float* B, float* out, int size_to_compute) {
+	for(int i = 0; i < size_to_compute; ++i) {
+		out[i] = A[i] + B[i];
+	}
+}
+inline PyObject* alt_op(char* trans, PyArrayObject* matrix) {
+	return (*trans == 'N' || *trans == 'n') ? (PyObject*)matrix : alt_transpose(matrix);
+}
+/* sgemm
+ * We assume that none of these 13 pointers passed as arguments is null.
+ * NB: We can more optimize this function (for example, when alpha == 0).
+ * */
+void sgemm_(char* TRANSA, char* TRANSB, 
+			const int* M, const int* N, const int* K,
```

Comment:
Don't use tab. Use 4 spaces indentation instead. We try to get rid of tab everywhere in Theano.